estimators/base_models.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+# Copyright 2022 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# """base_models!
+#
+# Base Models used by estimators.
+#
+# """
+
+import itertools
+import numpy as np
+from sklearn import ensemble
+from sklearn import linear_model
+from sklearn import metrics as sk_metrics
+import tensorflow as tf
+from tensorflow.keras import optimizers
+from tensorflow.keras import regularizers
+from tensorflow.keras import Sequential
+from tensorflow.keras.layers import Dense
+from tensorflow.keras.wrappers.scikit_learn import KerasRegressor
+
+
+def image_model_image_regression(model_config):
+    """Implements a one-layer NN that mimics a Linear Regression.
+
+    :param model_config: dictionary with parameters.
+    :return: model: tf.keras.Model.
+    """
+    # A simple linear regression implemented as NN.
+    last_activation = model_config.get('activation', 'linear')
+    image_size = [model_config['image_size'], model_config['image_size']]
+    inputs = tf.keras.Input(shape=[*image_size, 3], name='image_regression')
+    x = tf.keras.layers.Flatten()(inputs)
+    outputs = tf.keras.layers.Dense(1, activation=last_activation, use_bias=True,
+                                    kernel_regularizer=regularizers.l1_l2(
+                                        l1=1e-5,
```

Comment:
Better to accept these as model configurations.

Reply:
Done!